cpp/common/src/codingstandards/cpp/IntegerConstantMacro.qll

```diff
@@ -0,0 +1,37 @@
+import cpp
+
+/**
+ * The family of macros `xINTsize_C(arg)` (e.g. `UINT16_C(123)`) which are used
+ * to create an integer constant of type `Xint_leastSIZE_t` (e.g.
+ * `uint_least16_t).
+ */
+class IntegerConstantMacro extends Macro {
+  boolean signed;
+  int size;
+
+  IntegerConstantMacro() {
+    signed = true and size = getName().regexpCapture("INT(8|16|32|64)_C", 1).toInt()
+    or
+    signed = false and size = getName().regexpCapture("UINT(8|16|32|64)_C", 1).toInt()
+  }
+
+  predicate isSmall() { size < 32 }
```

Comment:
Use the size of the int type in this database - it's not guaranteed to be 32.
```suggestion
  predicate isSmall() { size <   any(IntType it | it.isSigned()).getSize() }
```

Reply:
Done!